cassandra/pool.py

```diff
@@ -572,14 +573,16 @@ def return_connection(self, connection):
             in_flight = connection.in_flight
 
         if connection.is_defunct or connection.is_closed:
```

Comment:
Yeah. That's unfortunate. I put this flagging and logic at the pool level to avoid Connection having to know about Cluster and pooling logic. They are not quite the same because of the difference around replacing connections.